### PR TITLE
DM-44414: Initial implementation of queryDatasets for RemoteButler

### PR DESCRIPTION
**Pull request by dhirving**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`
- [ ] (if changing dimensions.yaml) make a copy of dimensions.yaml in `configs/old_dimensions`

### CODE REVIEW

File: python/lsst/daf/butler/direct_query_driver/_result_page_converter.py [hunk 1]
```diff
@@ -129,12 +137,45 @@ def convert(
...
+        self._data_coordinate_converter = _DataCoordinateRowConverter(spec.dimensions, column_order)
+        self._column_order = column_order
+
+    def convert(
+        self,
+        raw_rows: Iterable[sqlalchemy.Row],
+        next_key: PageKey | None,
+    ) -> DatasetRefResultPage:
+        run_column = qt.ColumnSet.get_qualified_name(self._spec.dataset_type_name, "run")
+        dataset_id_column = qt.ColumnSet.get_qualified_name(self._spec.dataset_type_name, "dataset_id")
```
Comment (TallJimbo) [inline python/lsst/daf/butler/direct_query_driver/_result_page_converter.py:164]: I think these two names will need to get some name-shrinking applied.  I think everything else is safe.  I thought we already had a test that would flag this, but if not, I think we need to add one with a very long dataset type name.

File: python/lsst/daf/butler/remote_butler/registry/_query_datasets.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    """Implementation of the legacy ``DimensionRecordQueryResults`` interface
+    using the new query system.
+
+    Parameters
+    ----------
+    query_factory : `QueryFactory`
+        Function that can be called to access the new query system.
+    args : `CommonQueryArguments`
+        User-facing arguments forwarded from
+        ``registry.queryDimensionRecords``.
```
Comment (TallJimbo) [inline python/lsst/daf/butler/remote_butler/registry/_query_datasets.py]: Should be `queryDatasets`.

**Review by TallJimbo — APPROVED** [review]
Looks great.

I hope we can migrate our way out of the new `ingest_date` specialization in the near future.

File: python/lsst/daf/butler/direct_query_driver/_result_page_converter.py (hunk 1 above)
Comment (dhirving) [reply python/lsst/daf/butler/direct_query_driver/_result_page_converter.py:164]: Added a note to DM-44505 to deal with this... I'll check in with Andy on nameshrinker stuff and decide the best way to pass everything down to here.

queryDatasets test coverage is actually pretty sparse... there's a lot more cover of queryDataIds and I think it's assuming that most of the implementation is common between the two.

**Pull request merged**